pennylane/capture/dynamic_shapes.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains a utility for handling inputs with dynamically shaped arrays.
+"""
+from string import ascii_lowercase
+
+has_jax = True
+try:
+    import jax
+except ImportError:  # pragma: no cover
+    has_jax = False  # pragma: no cover
+
+
+def determine_abstracted_axes(args):
```

Comment:
I'm having a hard time reviewing this function because I'm not confident that I understand what the expected behaviour is for various inputs. Maybe we could have a quick call/meeting and you could walk me through it?

Reply:
I'll put in some more examples for how `abstracted_axes` work.